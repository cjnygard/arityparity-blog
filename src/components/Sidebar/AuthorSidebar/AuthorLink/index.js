// @flow strict
export { default } from './AuthorLink';
