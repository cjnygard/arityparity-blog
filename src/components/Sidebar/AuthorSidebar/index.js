// @flow strict
export { default } from './AuthorSidebar';
