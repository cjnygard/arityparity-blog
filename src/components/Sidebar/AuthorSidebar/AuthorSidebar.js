// @flow strict
import React from 'react';
import styles from './AuthorSidebar.module.scss';
import AuthorLink from './AuthorLink';

type Props = {
  authors: {
    name: string,
    bio: string,
    photo: string
  }[],
  isIndex: ?boolean
};

const AuthorSidebar = ({ authors, isIndex }: Props) => (
  <div className={styles['authorsidebar']}>
    <h3>Authors</h3>
    <ul className={styles['authorsidebar__list']}>
      {authors.map((author) => (
        <li className={styles['authorsidebar__list-item']} key={author.name}>
          <AuthorLink author={author} isIndex={isIndex} />
        </li>
      ))}
    </ul>
  </div>
);

export default AuthorSidebar;
