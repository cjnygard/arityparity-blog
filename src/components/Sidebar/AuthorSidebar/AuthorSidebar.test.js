// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import AuthorSidebar from './AuthorSidebar';

describe('AuthorSidebar', () => {
  const props = {
    author: {
      name: 'test',
      photo: '/carl-photo-social.jpg',
      bio: 'test'
    },
    isIndex: false
  };

  it('renders correctly', () => {
    const tree = renderer.create(<AuthorSidebar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
