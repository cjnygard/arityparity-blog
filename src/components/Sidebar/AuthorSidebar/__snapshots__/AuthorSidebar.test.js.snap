// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorSidebar renders correctly 1`] = `
<div
  className="authorsidebar"
>
  <h3>
    Authors
  </h3>
  <ul
    className="authorsidebar__list"
  >
    <li
      className="authorsidebar__list-item"
    >
      <ul
        className="author"
      >
        <li>
          <div
            className="author__headshot"
          >
            <img
              alt="test"
              className="author__headshot_photo"
              height="75"
              src="/carl-photo-social.jpg"
              width="75"
            />
          </div>
        </li>
        <li>
          <div
            className="author__name"
            height="75"
          >
            <a
              className="author__name-link"
              href="/pages/test/"
            >
              test
            </a>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
